### PR TITLE
 feat(cli): add "core.task_runner" and "core.enable_xcom_pickling" to unsupported config check to command "airflow config lint"

### LINKED ISSUE #45213: Airflow 2 to 3 auto migration rules - airflow config lint (44080 ~ 45017)
### Description

Parent issue: https://github.com/apache/airflow/issues/41641

### Removal
* `core.task_runner`
* `core.enable_xcom_pickling`

### Use case/motivation

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Lee-W**
## Why
Config `core.task_runner` and `core.enable_xcom_pickling` are removed and will no longer exist in airflow 3.0

## What
Add lint rules to check these 2 configs

Closes: #45213

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: airflow/cli/commands/remote_commands/config_command.py [hunk 1]
```diff
@@ -188,6 +190,9 @@ def message(self) -> str:
         config=ConfigParameter("core", "max_db_retries"),
         renamed_to=ConfigParameter("database", "max_db_retries"),
     ),
+    ConfigChange(config=ConfigParameter("core", "task_runner")),
+    ConfigChange(config=ConfigParameter("core", "enable_xcom_pickling")),
```
Comment (potiuk) [inline airflow/cli/commands/remote_commands/config_command.py:194]: Should we keep that one? I think we were going to move away from pickling Xcom?

Comment (amoghrajesh) [reply airflow/cli/commands/remote_commands/config_command.py:194]: Yeah, i think we did it here: https://github.com/apache/airflow/pull/43905.

**Review by amoghrajesh — APPROVED** [review]
LGTM +1 pending the clarification on xcom pickling, but I am sure you will handle that.
And thanks for code comments for each section, it helps!

File: airflow/cli/commands/remote_commands/config_command.py (hunk 1 above)
Comment (Lee-W) [reply airflow/cli/commands/remote_commands/config_command.py:194]: I'm a bit confused here. This PR was created to warn that `core.task_runner` and `core.enable_xcom_pickling` have been removed. Is there anything I missed or misunderstood?

Comment (amoghrajesh) [reply airflow/cli/commands/remote_commands/config_command.py:194]: Yeah, i think we should add the warning now that we move away from it.
@potiuk do you want any clarification here? I am good.

Comment (potiuk) [reply airflow/cli/commands/remote_commands/config_command.py:194]: Ah my bad.

**Pull request merged**